architecture.md

```diff
@@ -0,0 +1,78 @@
+This is the case for all libraries, whether their corresponding plugins affect the frontend or backend processes.
+
+The Ignition Physics library is only used in the physics plugin, not in other plugins nor in the core of Ignition Gazebo.
+Some libraries only have one corresponding plugin, or one in each process (frontend and backend).
+However, some libraries are used by all plugins, like Ignition Common which provides common functionality like logging, manipulating strings, file system interaction, etc., to all plugins.
+Other such libraries include Ignition Plugin, Ignition Math, SDFormat, and more.
+
+There are certain plugins in both the frontend and backend processes that are loaded by default in every version of Ignition.
+Many other plugins are optional.
+One common optional plugin is the sensors plugin.
+Both optional and default plugins can be removed or added at any time; Ignition Gazebo will continue to run with limited functionality.
+
+![Ignition Gazebo architecture diagram]()
+
+## Backend server process
+
+Ignition gazebo is responsible for loading plugins in the backend, referred to as systems.
+The server runs an entity-component system architecture (see [Ignition Gazebo terminology](https://ignitionrobotics.org/api/gazebo/4.2/terminology.html)).
+The backend will usually have multiple systems responsible for everything in the simulation – computing physics, recording logs, receiving user commands, etc.
+
+Systems act on entities and components of those entities.
+An entity is anything in the simulation scene (a model, link, light, actor, etc.), and components are their characteristics (pose, name, geometry, etc.).
+Server plugins have access to all entities and what components they have, and make modifications to those components.
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Libraries, however, don’t know about the entities and components.
+
+All of the systems in the backend create a loop, where some systems are proposing changes to entities and their components, and other systems are handling and applying those changes.
+This is called the “Simulation loop”.
+
+Physics, user commands, and scene broadcaster are all systems launched by default in the backend.
+As mentioned earlier, however, even default systems can be added to or removed from this loop.
+For any other functionality, like sensor data processing for example, a new system would have to be added.
+Plugins that generate sensor data and utilize the Ignition Sensors and Ignition Rendering libraries would have to be added to the backend process.
```

Comment:
For these two sentences, maybe "new" -> "additional"? Or reword?
Currently, it could sound like, we haven't developed such systems yet, and someone will have to write them. But we have a bunch non-default ones ready to be added.